multimodal/src/autogluon/multimodal/learners/default_learner.py

```diff
@@ -0,0 +1,1724 @@
+    extract_from_output,
+    filter_hyperparameters,
+    get_available_devices,
+    get_config,
+    get_detection_classes,
+    get_fit_complete_message,
+    get_fit_start_message,
+    get_local_pretrained_config_paths,
+    get_minmax_mode,
+    get_mixup,
+    get_stopping_threshold,
+    hyperparameter_tune,
+    infer_dtypes_by_model_names,
+    infer_metrics,
+    infer_precision,
+    infer_scarcity_mode_by_data_size,
+    init_df_preprocessor,
+    init_pretrained,
+    list_timm_models,
+    load_text_tokenizers,
+    logits_to_prob,
+    merge_bio_format,
+    modify_duplicate_model_names,
+    object_detection_data_to_df,
+    predict,
+    process_batch,
+    save_pretrained_model_configs,
+    save_result_df,
+    save_text_tokenizers,
+    select_model,
+    setup_detection_train_tuning_data,
+    setup_save_path,
+    split_hyperparameters,
+    tensor_to_ndarray,
+    turn_on_off_feature_column_info,
+    update_config_by_rules,
+    update_hyperparameters,
+    update_tabular_config_by_resources,
+    upgrade_config,
+)
+from .abstract_mm_learner import AbstractMMLearner
+
+
+logger = logging.getLogger(__name__)
+
+
+class DefaultLearner(AbstractMMLearner):
+    def __init__(
+        self,
+        label: Optional[str] = None,
+        problem_type: Optional[str] = None,
+        column_types: Optional[dict] = None,
+        # query: Optional[Union[str, List[str]]] = None,
+        # response: Optional[Union[str, List[str]]] = None,
+        # match_label: Optional[Union[int, str]] = None,
+        # pipeline: Optional[str] = None,
+        # presets: Optional[str] = None,
+        eval_metric: Optional[str] = None,
+        path: Optional[str] = None,
+        verbosity: Optional[int] = 2,
+        num_classes: Optional[int] = None,
+        classes: Optional[list] = None,
+        enable_progress_bar: Optional[bool] = None,
+    ):
+        super().__init__()
+
+        self._label_column = label
+        self._problem_type = problem_type
+        # self._presets = presets.lower() if presets else None  # NOTE: This will be handled by predictor.
+        self._eval_metric_name = eval_metric
+        self._validation_metric_name = None
+        self._output_shape = num_classes
+        self._classes = classes  # TODO: To be refactored into detection learner only
+
+        self._ckpt_path = None
+        self._pretrained_path = None
+        self._config = None
+        self._df_preprocessor = None  # initialize it as None.
+        self._column_types = column_types  # predictor will setup column_types, and set values here.
+        self._data_processors = None  # initialize it as None.
+        self._model_postprocess_fn = None
+        self._model = None
+        # self._resume = False  # This is for predictor to handle.
+        self._verbosity = verbosity
+        # self._warn_if_exist = warn_if_exist  # unused
+        self._enable_progress_bar = enable_progress_bar if enable_progress_bar is not None else True
```

Comment:
Why not just have `enable_progress_bar=True` as the default? It does the same thing as this line, but is easier to read

Reply:
Agreed!